### DISCUSSION TITLE
How to use grun with Python grammar

### DISCUSSION THREAD

**Discussion by juli1** [antlr/antlr4#3828] [Q&A] [answered]
Hello,

I am currently working with the [python grammar](https://github.com/antlr/grammars-v4/tree/master/python) and I would like to use `grun` on it. However, when running `grun` on it, I get an exception (see below).

Is there any guidance on how to use `grun` on the Python grammar?

```
$ grun PythonParser -gui
Exception in thread "main" java.lang.ClassCastException: class PythonParser
	at java.base/java.lang.Class.asSubclass(Class.java:3640)
	at org.antlr.v4.gui.TestRig.process(TestRig.java:135)
	at org.antlr.v4.gui.TestRig.main(TestRig.java:119)
```

**Comment by kaby76** [comment, marked as answer]
```
git clone https://github.com/antlr/grammars-v4
cd grammars-v4/python/python
mkdir x
cp *.g4 x
cp Java/* x
cd x
java -jar ~/.m2/antlr4-4.10.1-complete.jar *.g4
javac -cp ~/.m2/antlr4-4.10.1-complete.jar:. *.java
java -cp ~/.m2/antlr4-4.10.1-complete.jar:. org.antlr.v4.gui.TestRig Python root ../examples/argument.py -gui
```
You may have to play around with the classpath command-line option depending on Linux or Windows.

**Reply by juli1** [reply]
Thank you! It was `Python` and not `PythonParser`! Thank you so much!
